lib/exirc/exirc.ex

```diff
@@ -29,8 +29,24 @@ defmodule ExIRC do
       ExIRC.Client.stop! client
 
   """
-  use Supervisor
-  import Supervisor.Spec
+  use DynamicSupervisor
+
+  defmodule TemporaryClient do
```

Comment:
This allows us to define the module differently if it's started using the dynamic supervisor. There may be a few ways to do this but this seemed the most clear currently.